Error: Can't resolve 'webpack-dev-server/client?http://0.0.0.0:0' [1.7]

### Versions
```
Angular CLI: 1.7.0
Node: 8.9.4
OS: linux x64
Angular: 5.2.5
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cdk: 5.2.1
@angular/cli: 1.7.0
@angular/flex-layout: 2.0.0-beta.12
@angular/material: 5.2.1
@angular-devkit/build-optimizer: 0.3.1
@angular-devkit/core: 0.3.1
@angular-devkit/schematics: 0.3.1
@ngtools/json-schema: 1.2.0
@schematics/angular: 0.3.1
@schematics/package-update: 0.3.1
typescript: 2.4.2
webpack-node-externals: 1.6.0
webpack: 3.10.0
```

### Repro steps
Havn't got a repro sorry :/


### Observed behavior
```
ERROR in multi webpack-dev-server/client?http://0.0.0.0:0 ./path/to/main.ts
Module not found: Error: Can't resolve 'webpack-dev-server/client?http://0.0.0.0:0' in '/home/f.wiles/Documents/projectname'
```


### Mention any other details that might be useful (optional)
This is a medium sized project using `yarn` as the package manager
```
$ npm ls webpack-dev-server
projectname@0.0.0 /home/f.wiles/Documents/project-name
└─┬ @angular/cli@1.7.0
     └── webpack-dev-server@2.11.1
```
I've also tried deleting both `yarn.lock` and `node_modules` but the same error is observed

--- a/packages/@angular/cli/tasks/serve.ts
+++ b/packages/@angular/cli/tasks/serve.ts
@@ -109,8 +109,14 @@ export default Task.extend({
     if (serveTaskOptions.liveReload) {
       // This allows for live reload of page when changes are made to repo.
       // https://webpack.js.org/configuration/dev-server/#devserver-inline
+      let webpackDevServerPath;
+      try {
+        webpackDevServerPath = require.resolve('webpack-dev-server/client');
+      } catch {
+        throw new SilentError('The "webpack-dev-server" package could not be found.');
+      }
       let entryPoints = [
-        `webpack-dev-server/client?${clientAddress}`
+        `${webpackDevServerPath}?${clientAddress}`
       ];
       if (serveTaskOptions.hmr) {
         const webpackHmrLink = 'https://webpack.js.org/guides/hot-module-replacement';

